src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h

```diff
@@ -67,7 +67,7 @@ namespace Aws
          */
         struct AWS_CORE_API ClientConfiguration
         {
-            ClientConfiguration();
+            ClientConfiguration(bool shouldDisableIMDS = false);
```

Comment:
Ok so as discussed this will not compile because

```
explicit ClientConfiguration(bool useSmartDefaults, 
  const char* defaultMode = "legacy", 
  bool shouldDisableIMDS = false);
```

resolves ambiguously to this. One way around this is that we could be creating a `DefaultConfiguration` struct that can be passed optionally to the default constructor that we can add things to in over time i.e.

```
struct DefaultClientConfiguration {
  bool shouldDisableIMDS = false;
};

struct ClientConfiguration {
  ...
  ClientConfiguration(const DefaultClientConfiguration &configuration = {})
  ...
}
```

then invoked though

```
ClientConfiguration::ClientConfiguration(const DefaultClientConfiguration &configuration)
{
    this->disableIMDS = configuration.shouldDisableIMDS;
    ...
}
```

this was we dont have to add more arguments each time we want to add something to the default constructor _and_ it doesnt resolve ambigiously